### PR TITLE
add flexible ns to grafana dashboard test

### PR DESCRIPTION
**Pull request by whitneygriffith**
**Please provide a description of this PR:**

This allows the Grafana sample to be deployed in the provided `TelemetryNamespace`. Also, updated the VirtualService and Gateway to use v1.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
Just curious, why cannot you test in `istio-system`?

I get not wanting to install Istio there, but why can you not install _anything_ there?

**Comment by whitneygriffith** [comment]
> Just curious, why cannot you test in `istio-system`?
It will require managing the creation and deletion of `istio-system` ns for the sake of installing testing artifacts. When the typical flow for a user not using `istio-system` is to set their existing ns for `systemNamespace` and/or `TelemetryNamespace` with the expectation the tests will deploy any testing artifacts to that existing ns.

**Comment by howardjohn** [comment]
You also need to create/delete the `TelemetryNamespace`, why does it matter if it is istio-system?

**Comment by howardjohn** [comment]
I am not super against it, just might make things easier for everyone

**Comment by whitneygriffith** [comment]
> You also need to create/delete the `TelemetryNamespace`, why does it matter if it is istio-system?

The user wouldn't deploy the Telemetry Ns just for tests. If their setup includes a ns that is not `istio-system` for deploying Istio/Telemetry related stuff, the management of Telemetry Ns will already be baked into their process. 

By [default](https://github.com/istio/istio/blob/f0e8a11fdad450d94b434bcd38d0c68e96983a8b/pkg/test/framework/components/istio/config.go#L95) `TelemetryNamespace` is set to `DefaultSystemNamespace` which is `istio-system` so a user that does not care about using non `istio-system` will not be impacted by this change.

**Comment by howardjohn** [comment]
I understand its continuing the pattern, I just don't understand why someone would care. You cannot pick the namespace of `echo-<random suffix>` for instance, so why do you need to pick the name of this?

**Comment by whitneygriffith** [comment]
> I understand its continuing the pattern, I just don't understand why someone would care. You cannot pick the namespace of `echo-<random suffix>` for instance, so why do you need to pick the name of this?

If it is the `systemNamespace` being used here instead of `TelemetryNamespace` does that alleviate your concern?

**Comment by howardjohn** [comment]
Why not just have it create istio-system if it does not exist then?

**Comment by whitneygriffith** [comment]
> Why not just have it create istio-system if it does not exist then?

Are you proposing adopting this across the test suite? Where for all resources created for testing purposes (zipkin, grafana, prometheus, etc.) will be deployed to `istio-system`? If that's the case will we then remove the concept of `TelemetryNamespace` and `SystemNamespace`?

I am not opposed to either approach, I am just not sure if there is something truly wrong with the existing strategy of giving the user the choice to dictate target ns using `SystemNamespace` and `TelemetryNamespace` flags for us to warrant reverting our strategy.

**Comment by nshankar13** [comment]
Also working with Whitney to look into some of the Telemetry e2es. Would it make sense to keep with the existing pattern in tests/documentations of deploying applications (echo, etc) in their own namespace, and deploying telemetry Pods / Services in the root / system namespace? (whether `istio-system` or something else), for consistency?

File: tests/integration/telemetry/api/dashboard_test.go [hunk 1]
```diff
@@ -145,11 +145,15 @@ func TestDashboard(t *testing.T) {
...
 
 			// Apply just the grafana dashboards
-			cfg, err := os.ReadFile(filepath.Join(env.IstioSrc, "samples/addons/grafana.yaml"))
+			cfgBytes, err := os.ReadFile(filepath.Join(env.IstioSrc, "samples/addons/grafana.yaml"))
 			if err != nil {
 				t.Fatal(err)
 			}
-			t.ConfigKube().YAML("istio-system", yml.SplitYamlByKind(string(cfg))["ConfigMap"]).ApplyOrFail(t)
+			cfg := string(cfgBytes)
+			if ist.Settings().TelemetryNamespace != "istio-system" {
```
Comment (MorrisLaw) [inline tests/integration/telemetry/api/dashboard_test.go:153]: If we do decide to keep this logic, then we should add a comment highlighting the points you made when responding to john [here](https://github.com/istio/istio/pull/52062#issuecomment-2228829749).

**Pull request closed without merging**